exareme2/controller/services/exaflow/algorithm_flow_engine_interface.py

```diff
@@ -23,7 +20,7 @@ def __init__(
     def run_algorithm_udf(self, func, positional_args) -> List[dict]:
         tasks = []
         for task_handler in self._tasks_handlers:
-            udf_name = exaflow_registry.resolve_key(func)
+            udf_name = make_unique_func_name(func)
```

Comment:
It's not a "unique" function name actually, the name is a bit misleading.

And it's not a `udf_name` also...

Why not send the algorithm name and the udf name, and let the worker create the relative name?

Otherwise:
1) udf_name -> udf_registry_key
2) make_unique_func_name -> get_registry_key


But I think it would be better to pass the algorithm_name/udf_name and let the worker do this. Otherwise we are mingling controller/worker logic through task calls.